packages/relay/src/lib/types/RequestDetails.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/**
+ * Interface representing the details of a request.
+ */
+export interface IRequestDetails {
+  /**
+   * The unique identifier for the request.
+   * @type {string}
+   */
+  requestId: string;
+
+  /**
+   * The IP address from which the request originated.
+   * @type {string}
+   */
+  ipAddress: string;
+
+  /**
+   * The connection ID associated with the request (optional).
+   * @type {string | undefined}
+   */
+  connectionId?: string;
+}
+
+/**
+ * Represents the details of a request.
+ */
+export class RequestDetails {
```

Comment:
Just a suggestion, thinking out loud — Personally, since RequestDetails is a full class definition, which isn't typically something I'd expect to find in a `types/` folder.

I would recommend the following changes:
1. Move the `RequestDetails` class to a more appropriate location, such as a `models` or `entities` folder (`packages/relay/src/lib/models/` or `packages/relay/src/lib/entities/`).
2. Keep the `IRequestDetails` interface in the `types/` folder and rename the file to `IRequestDetails.ts`.

However, this would involve updating all import statements, given that this file impacts the entire codebase. I'd suggest handling this update in a separate PR, as the current one is already quite large.

Reply:
Okay, may be addressed in a separate PR